Api fetcher

pgicking opened the pull request:
Got some basic api calling functionality in here. 

The highways and highway classes might need to go in their own sub folders

There is one generic class that calls the api and returns what it gets as a json object. Although josh might have some suggestions on other ways to go about this. Might not be the program flow we end with but we can get all highways that we need to start. 

Also i had to move the swiftJSON.swift code into the main directory tree as theres a bug of some kind in swiftjson that causes segfaults when you try to define JSON as a return type, you can read about it here

pgicking committed:
Updated readme, installed swiftJson

pgicking committed:
Loc wants the JSON framework pushed :P

pgicking committed:
including podfile

locdle committed:
Merge branch 'ApiFetcher'

pgicking committed:
Figured out where i find out what api call returns

pgicking committed:
successful simple call against api

pgicking committed:
Switching to synchronous requests because asynchronous was being difficult

pgicking committed:
Bug in swiftyJSON causes segfault when returning JSON as a type, had to move swiftJSON.swift in the main directory tree to fix...

pgicking committed:
Added highways/highway class

pgicking committed:
added constructor for highway that requires a son object to fill it, this could be a questionable design choice :P

pgicking committed:
Remove call to fetch all in app delegate

locdle committed:
add xcworkspace files

locdle committed:
update gitignore to ignore some pod autogenerations files

locdle commented:
look good to me :), but I'm scared to merge the big  PR like that. @joshproehl can you double check?

joshproehl committed:
Podfile.lock should not be ignored.

http://guides.cocoapods.org/using/using-cocoapods.html#what-is-a-podfilelock

joshproehl committed:
Re-add all pods files, remove .gitignore

I've never used cocoapods where the Pods directory was checked in,
usually it's just the podfile and the podfile.lock...
I've been doing some research to try and figure out how this best works,
and it appears that checking in the Pods/ directory should be an
all-or-nothing deal.... So I've gone ahead and added everything that was
generated by the pod update here, and removed it from the .gitignore. I
*think* that should allow anybody who picks up the project to "just go"
without having to mess with updating pods.

I kept this as a seperate commit from the last one because the .lock
file should DEFINITELY be in source control, whichever way this commit
goes down, so it makes it easy to revert this if necessary.

joshproehl commented:
I picked out all the code not related to cocoapods into a new branch, and added swiftyjson as a single file rather than a cocoapod disaster. :grinning: 

master, as of 482681622addb5bf2477af6bbdba1f22c24c5feb, should have all changes from this PR.
